pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -1668,12 +1668,8 @@ func (r *ReconcilePerconaServerMongoDB) createOrUpdate(ctx context.Context, obj
 
 	updateObject := false
 	if oldObject.GetAnnotations()["percona.com/last-config-hash"] != hash ||
-		!compareMaps(oldObject.GetLabels(), obj.GetLabels()) {
+		!compareMaps(oldObject.GetLabels(), obj.GetLabels()) || !compareMaps(oldObject.GetAnnotations(), obj.GetAnnotations()) {
```

Comment:
why do we bring back this check? it'll introduce a regression, see: https://forums.percona.com/t/is-exposing-mongo-replicaset-with-nodeport-working/11827/25